src/utils/common.cpp

```diff
@@ -23,3 +23,34 @@ std::string get_current_git_path()
 // sub->add_option("directory", directory, "info about directory arg")
 //     ->check(CLI::ExistingDirectory | CLI::NonexistentPath)
 //     ->default_val(std::filesystem::current_path());
+
+git_strarray_wrapper::git_strarray_wrapper(std::vector<std::string> m_patterns)
+    : m_patterns(std::move(m_patterns))
+{
+    init_str_array();
+}
+
+git_strarray_wrapper::git_strarray_wrapper(git_strarray_wrapper&& rhs)
+    : m_patterns(std::move(rhs.m_patterns))
+{
+    init_str_array();
```

Comment:
A good practice is to let rhs in a "valid" state. To do so, this constructor can call a `reset_str_array`  method (see the comment for the destructor).